contracts/lib/LibPerpetuals.sol

```diff
@@ -28,6 +28,29 @@ library Perpetuals {
         return keccak256(abi.encode(order));
     }
 
+    /**
+     * @return An updated average execution price, based on previous rolling average, and new average
+     * @param oldFilledAmount The filled amount that will be getting changed
+     * @param oldAverage The average rolling execution price that will be updated
+     * @param fillChange The amount of units being added to the filledAmount
+     * @param newFillExecutionPrice The execution price of the fillChange units
+     */
+    function calculateAverageExecutionPrice(
+        uint256 oldFilledAmount,
+        uint256 oldAverage,
+        uint256 fillChange,
+        uint256 newFillExecutionPrice
+    ) internal pure returns (uint256) {
+        uint256 oldFactor = PRBMathUD60x18.mul(oldFilledAmount, oldAverage);
+        uint256 newFactor = PRBMathUD60x18.mul(fillChange, newFillExecutionPrice);
+        uint256 newTotalAmount = oldFilledAmount + fillChange;
+        if (newTotalAmount == 0) {
+            return 0;
+        }
+        uint256 average = PRBMathUD60x18.div(oldFactor + newFactor, newTotalAmount);
+        return average;
+    }
+
```

Comment:
Should this be in `LibPerpetuals` or our internal `LibMath`? Realistically all it does is updates a running mean (as the doc comment states).

Reply:
I personally feel like it's more suited to `LibPerpetuals` just because it's a bit more context-focused. As in, while it is a pretty mathsy function, it's use-case in our situation is specific to order execution if you get what I'm saying. But I'm not too fussed, if you disagree I'll chuck it in LibMath.